### DISCUSSION TITLE
High CPU on high frame rate on 11.0 vs 0.10

### DISCUSSION THREAD

**Discussion by ahopper** [AvaloniaUI/Avalonia#10875] [General]
My app that often runs with parts of the screen updating at 60fps is using 4x cpu on the current master vs 0.10.18 , this is on windows 10 on an average intel I5. If I set UseWgl=false, AllowEglInitialization=false the cpu halves but is still 2x as high as 0.10.18. Most of the updates are writeablebitmaps changing.  I've only just started investigating this so maybe I'm just doing something that is not 11.0 friendly. Just wondered if anyone else was looking at or experiencing this.

**Comment by mysteryx93** [comment]
What I did notice is that when performing CPU-intensive background work (i.e. encoding), the UI doesn't update well and can update once per second or less. Dunno if related and I haven't investigated further.

**Comment by ahopper** [comment]
Here is a repro 
[AvaloniaVSync.zip](https://github.com/AvaloniaUI/Avalonia/files/11132404/AvaloniaVSync.zip)
This uses 24% total cpu on master and 11% on 0.10.18 on windows 10.

**Comment by ahopper** [comment]
On profiling it appears most time is spent in d3d10warp.dll, should this be being used?
AllowEglInitialization=false speeds up the repro and then warp does not appear to be used.
I see this on 3 different computers so I don't think it is just an unlucky setup.

**Reply by Gillibald** [reply]
Warp is the DirectX software rasterizer that is being used if no graphics hardware is present

**Comment by ahopper** [comment]
I discovered one computer had its display adapter disabled, a bios update fixed it and it now runs much faster, however another computer without the display adapter issue still shows the slow down.

**Comment by ahopper** [comment]
On the computer which now behaves I noticed that setting UseWgl=true halves the cpu

**Comment by Coloryr** [comment]
This is my program test.
```
<PackageReference Include="Avalonia" Version="11.0.0-preview6" />
```
![3U`YC_X443ZXQS22J04R@2L](https://user-images.githubusercontent.com/26276037/229484736-6694833b-c7ae-4f2c-9a32-6de7b9c1e514.png)
![image](https://user-images.githubusercontent.com/26276037/229484776-2f0ea08e-1895-4e3f-849b-77ee78eed4f4.png)

Only open the window & not select.

```
AllowEglInitialization: default
UseWindowsUIComposition: default
UseWgl: default
```
![image](https://user-images.githubusercontent.com/26276037/229484913-c1c1b6a3-8361-4044-9f7c-15b21352c698.png)

```
AllowEglInitialization: false
UseWindowsUIComposition: false
UseWgl: false
```
![image](https://user-images.githubusercontent.com/26276037/229485152-437651f9-c659-4c34-a1e1-0bebc94c318c.png)

```
AllowEglInitialization: default
UseWindowsUIComposition: default
UseWgl: true
```
![image](https://user-images.githubusercontent.com/26276037/229485290-a8e98527-acd7-4081-975a-9eabe389749f.png)

```
AllowEglInitialization: default
UseWindowsUIComposition: false
UseWgl: default
```
![image](https://user-images.githubusercontent.com/26276037/229485512-6ce73c0b-0836-476e-a15b-9db918be47c7.png)

```
AllowEglInitialization: false
UseWindowsUIComposition: default
UseWgl: default
```
![image](https://user-images.githubusercontent.com/26276037/229485689-2f46aea1-9ede-46c3-aae8-4a5fd10d39c0.png)

**Comment by ahopper** [comment]
I'm beginning to wonder if I had two machines with disabled gpu, the second one now also does not demonstrate this issue ( I can't explain what has changed but it did update between tests). The third report of this was from a user with a fairly low end windows tablet, I shall try and find out what the state of their gpu is.  It appears falling back to warp is worse than the AllowEglInitialization=false option when no gpu.

**Comment by ahopper** [comment]
once the gpu worked the best settings I found were just UseWindowsUIComposition=false this uses similar %cpu to UseWgl=true but has better vsync
